spec/widgets-service.spec.js

```diff
@@ -0,0 +1,274 @@
+var specHelper = require('./spec-helper');
+var WidgetsService = require('../src/widgets-service');
+var WidgetsCollection = require('../src/widgets/widgets-collection');
+
+describe('widgets-service', function () {
+  beforeEach(function () {
+    this.vis = specHelper.createDefaultVis();
+    this.widgetsCollection = new WidgetsCollection();
+    this.widgetsService = new WidgetsService(this.widgetsCollection, this.vis.dataviews);
+  });
+
+  it('should return the WidgetsService instance', function () {
+    expect(this.widgetsService).toBeDefined();
+  });
+
+  describe('.get', function () {
+    it('should return the corresponding widgetModel for given id', function () {
+      expect(this.widgetsService.get('some-id')).toBeUndefined();
+
+      var aWidgetModel = this.widgetsCollection.add({
+        id: 'some-id'
+      });
+      expect(this.widgetsService.get('some-id')).toBe(aWidgetModel);
+    });
+  });
+
+  describe('.newCategoryModel', function () {
+    describe('when given valid input', function () {
+      beforeEach(function () {
+        var attrs = {
+          title: 'some_title',
+          column: 'my_column',
+          aggregation: 'avg'
+        };
+        this.results = this.widgetsService.newCategoryModel(attrs, this.vis.map.layers.first());
```

Comment:
In order to make this test a bit more clear, we could rename `this.results` to `this.widgetModel`.